core/FrontController.php

```diff
@@ -430,6 +430,7 @@ public function init()
         ) {
             Request::reloadAuthUsingTokenAuth();
             Request::checkTokenAuthIsNotLimited($module, $action);
+            Session::close();
```

Comment:
Any reason for closing the session at this point?

Reply:
During the process somewhere we start it, but actually we don't need it as we handle this request as stateless, so I thought we can close it (the original idea was coming from @tsteur).

Reply:
In theory that might make sense, but not sure if that could cause any regressions if someone tries to perform a controller action with a token_auth, that actually tries to change the session. But maybe we can simply change it and wait if any regressions are reported.

Reply:
It should be fine to close the session if `force_api_session` is not present in the URL AFAIK. There could be a regression but it would be hard to know. In theory though in these cases the session shouldn't be started anyway.

Reply:
Would this be an issue for the "embed all of matomo" use case (one big client uses this, can't remember their name starts w/ an 'M')? Ie, the `[General] enable_framed_pages` `[General] enable_framed_settings` config. I would think the session might be written to there.

Reply:
Also, thought it might be useful to have test pages for these, so created: https://github.com/matomo-org/test-examples/pull/6

Reply:
I'm not familiar with those settings. I tried with the new test-example and worked. How else should we test it?

Reply:
They allow users to embed whole matomo pages. One company I know of uses it to apply their own branding to all of Matomo and embed it as part of their website for their customers.

I think to test that it would work we'd need to write the session when a page is framed and see if it persists? Maybe sending a notification from PHP and seeing it in the embedded admin page?

Reply:
I tried something.

1. created a persistent notification in a controller
2. uncommented the line where I created it
3. reloaded Matomo page, the notification was there

I did the same when Matomo was running inside an iframe. This time at step 3, the notification was not there. It means the session was not persisted anyway.

What is the conclusion here? Should we keep that `Session::close()` line?
ping @tsteur @diosmosis @sgiehl

Reply:
done 👍

Reply:
If the request is an API call then we for sure don't want to use any session but widgetized requests may use a session I would say

Reply:
@flamisz I think ^ means my idea wouldn't be enough